gui/wxpython/gui_core/pyedit.py

```diff
@@ -269,6 +269,22 @@ def __init__(self, panel, guiparent, giface):
         self.overwrite = False
         self.parameters = None
 
+        self.guiparent.recent_files.recent.Bind(
+            wx.EVT_MENU_RANGE, self.OnFileHistory,
+            id=wx.ID_FILE1,
+            id2=wx.ID_FILE + self.guiparent.recent_files.history_len,
+        )
+
+    def OnFileHistory(self, event):
+        """Choose recent file from menu and read file content"""
+        file_num = event.GetId() - wx.ID_FILE1
+        path = self.guiparent.recent_files.filehistory.\
+            GetHistoryFile(file_num)
+        self.filename = path
+        with open(path, 'r') as f:
+            self.body.SetText(f.read())
```

Comment:
On the other hand, this should be in the controller (as all the other file opening cases are I assume). This of course makes too separated, but that exactly where the signals can help. You do the bind to OnFileHistory in the RecentFilesMenu class (function can be private), but you emit the signal there. In the user code, i.e., the controller, you connect to that signal instead of using the Bind on the menu object which is (should be) internal to the RecentFilesMenu.